hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/sigs/HollowScreening.java

```diff
@@ -0,0 +1,183 @@
+     *
+     * @param pendingCompletions a list of {@link PendingCompletion} if there the hollow screening found possible
+     *                           hollow account completions. {@code null} otherwise
+     * @param replacedPayerKey a {@link JECDSASecp256k1Key} if payer key was {@link JWildcardECDSAKey} and a corresponding
+     * {@link JECDSASecp256k1Key} was found in the txn sigs. {@code null} otherwise
+     * @param replacedOtherKeys a list of all other req keys, where possible {@link JWildcardECDSAKey} -> {@link JECDSASecp256k1Key}
+     *                          conversions were made; {@code null} if no conversions were made
+     */
+    record HollowScreenResult(
+            @Nullable List<PendingCompletion> pendingCompletions,
+            @Nullable JKey replacedPayerKey,
+            @Nullable List<JKey> replacedOtherKeys) {}
+
+    /**
+     * <strong>NOTE</strong>- should be called only if {@link HollowScreening#atLeastOneWildcardECDSAKeyIn} for the same
+     * {@code payerKey} and {@code otherKeys} returned true
+     *
+     * For each {@link JWildcardECDSAKey} in {@code payerKey} and {@code otherKeys}, tries to find its corresponding
+     * {@link JECDSASecp256k1Key} from the {@code txnSigs}, and also construct a {@link PendingCompletion} if a hollow
+     * account is linked to the evm address from the {@link JWildcardECDSAKey};
+     *
+     * <p>Note that this method tries to replace {@link JWildcardECDSAKey}s present in {@code otherKeys}
+     * even if no pending completions are present. That's because a CryptoCreate with an evm address alias, derived from
+     * a key, different than the key being set for the account, also adds a {@link JWildcardECDSAKey} to the {@code otherKeys},
+     *  <strong>but that {@link JWildcardECDSAKey} may not be connected to a finalization.</strong>
+     *
+     * @param txnSigs the list of transaction signatures
+     * @param payerKey the payer key of the current txn
+     * @param otherKeys the other req keys for the current txn
+     * @param aliasManager an alias resolver
+     * @return an instance of {@link HollowScreenResult}
+     */
+    static HollowScreenResult performFor(
+            @NonNull final List<TransactionSignature> txnSigs,
+            @NonNull final JKey payerKey,
+            @NonNull final List<JKey> otherKeys,
+            @NonNull final AliasManager aliasManager) {
+        final var evmAddressToEcdsaKeyIndex = createEvmAddressToEcdsaKeyIndexFrom(txnSigs);
+
+        List<PendingCompletion> pendingCompletions = null;
+        JKey replacedPayerKey = null;
+        if (payerKey.hasWildcardECDSAKey()) {
+            final var hollowKey = payerKey.getWildcardECDSAKey();
+            final var payerEvmAddress = hollowKey.getEvmAddress();
+            final var correspondingKey = evmAddressToEcdsaKeyIndex.get(Bytes.of(payerEvmAddress));
+            if (correspondingKey != null) {
+                replacedPayerKey = correspondingKey;
```

Comment:
If `correspondingKey == null` here, then we will fail with `INVALID_PAYER_SIGNATURE`, right? That might be worth adding a comment about.

Reply:
Correct, will add